viteup/src/__tests__/derive-entrypoints.spec.ts

```diff
@@ -541,4 +541,36 @@ describe("deriveEntrypoints", () => {
       );
     });
   });
+
+  describe("when provided a standard package.json with multiple type of exports: one export w/ types + default conditions, the package json export, one css export and one folder exports", () => {
+    it("derives exports", () => {
+      expect(
+        deriveEntrypoints(
+          {
+            module: true,
+            outDir: "dist",
+          },
+          {
+            "./package.json": "./package.json",
+            ".": {
+              types: "./dist/index.d.ts",
+              source: "./src/index.ts",
+              default: "./dist/index.mjs",
+            },
+            "./goo": {
+              types: "./dist/goo/index.d.ts",
+              source: "./src/goo/index.ts",
+              default: "./dist/goo/index.mjs",
+            },
+            "./foo/styles.css": "./dist/foo/styles.css",
```

Comment:
In our use case, the output path of this export entry is not in the `dist` output directory.

In the future, exports such as `"./foo/styles.css": "./dist/foo/styles.css"` will need to be treated as entry points, as CSS compilation is a common use case & it's supported by Vite afaik.